Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm

```diff
@@ -12317,7 +12317,23 @@ - (void)continueContextMenuInteraction:(void(^)(UIContextMenuConfiguration *))co
 
 #if ENABLE(DATA_DETECTION)
     if ([(NSURL *)linkURL iTunesStoreURL]) {
-        [self continueContextMenuInteractionWithDataDetectors:continueWithContextMenuConfiguration];
+        bool shouldContinueContextMenuInteraction = [&] {
+            if (!_page->websiteDataStore().isPersistent())
+                return false;
+
+#if ENABLE(NETWORK_CONNECTION_INTEGRITY)
+            if (_page->networkConnectionIntegrityPolicies().contains(WebCore::NetworkConnectionIntegrity::Enabled))
+                return false;
+#endif
+
+            return true;
+        }();
+
+        if (shouldContinueContextMenuInteraction)
+            [self continueContextMenuInteractionWithDataDetectors:continueWithContextMenuConfiguration];
+        else
+            continueWithContextMenuConfiguration(nil);
+
```

Comment:
Nit - extra newline.

Reply:
is this really an extra newline? I did it intentionally since otherwise its immediately after the `else` branch which makes it harder to read